From bab183d6357dfaee99a0e9666246a90d1f960592 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Fri, 26 Mar 2021 11:31:54 +0100
Subject: GCC: allow to disable NEON with GN.

In case NEON support is broken for GCC, we want to be able to disable
it temporarilly. There is a GN flag libyuv_use_neon that should allow
that.

But, when set to false, libyuv_internal elements are still trying to
link to NEON implementation. We should pass LIBYUV_DISABLE_NEON
define to avoid that.

Bug: 819294
---
 BUILD.gn | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/BUILD.gn b/BUILD.gn
index e1c7c1da..c01d02f3 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -156,6 +156,10 @@ static_library("libyuv_internal") {
     deps += [ "//third_party:jpeg_includes" ]
   }
 
+  if (!libyuv_use_neon) {
+    defines += [ "LIBYUV_DISABLE_NEON" ]
+  }
+
   # Always enable optimization for Release and NaCl builds (to workaround
   # crbug.com/538243).
   if (!is_debug || is_nacl) {
-- 
2.25.1

